### ISSUE TITLE
Fileupload Qcubed V4

### ISSUE THREAD

**Issue by RD1612** [qcubed/qcubed#1300]
Hi, 
Does anybody know how i can do a fileupload in Qcubed V4 ? 
In V3 i used a FileAsset, but in the new version it doesn't seem to work.

Thanks in advance 

Kenny

**Comment by RD1612** [comment]
I tried with the class FileControl also but when i do a form post the $_FILES array is empty.
below is the code that i use to generate the upload field.
`$this->flDocument				= new \QCubed\Control\FileControl($this);
$this->btnSave					= \QCubed\Project\Control\Button::GetSaveButton($this, tr('Save'));`

The page has only this field and a save button (for testing purposes) 
I'm brand new to the Qcubed framework and for the moment I'm a bit lost.
If someone can help me, it would be really appreciated 
If you need some additional code just let me know.

**Comment by kukrik** [comment]
Yes, unfortunately, is the case. Fileupload is so old and has been a bit behind the times.

This FileControl class will render an HTML File input and no more... It is necessary to build an application on top.

I have a plan to make a fileupload plugin, but it will take some more time. Because I need to use it repeatedly.

As I have somewhat understood, the @spekary also has a plan to do a fileupload. 

We have to ask him.

**Comment by matthiaz** [comment]
I'm guessing GetSaveButton does something like 
`$btn->addAction(new \QCubed\Event\Click(), new \QCubed\Action\AjaxControl($this, 'Save'));`

Change that to 
`$btn->addAction(new \QCubed\Event\Click(), new \QCubed\Action\ServerControl($this, 'Save'));`

When posting files you need to use a server control, the current file upload thingy doesn't work through AjaxControl.

**Comment by spekary** [comment]
I have no immediate plans. I have looked at it a couple of times. There are multiple ways to do it in Ajax, and so I couldn't come up with a nice general way of doing it. But I may look at it again. I noticed a recent change in some browsers where you cannot just drag a file over the Choose File button to upload it. That adds and additional couple of steps to doing an upload, so its now sufficiently annoying to warrant looking at it again. Also, there are now enough browsers supporting drag and drop that this might be sufficiently generally doable without a lot of special purpose code.

Note that the behavior on phone browsers is quite different. Most phone browsers have the ability to select images, or even capture an image when you click a choose file button. That will need to be taken into account.
